File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void traversal(TreeNode root, List<Integer> list,int level)
    {
        if(root==null) return;

        if(level==list.size())
            list.add(root.val);

        traversal(root.right,list,level+1);
        traversal(root.left,list,level+1);
    
    }
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> list = new ArrayList<>();
        traversal(root, list,0);
        return list;
    }
}

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = [] 
        def dfsTraversal(node = root , level = 0):
	        if not node:
		        return
	        if level == len(res):
		        res.append(node.val)
	        dfsTraversal(node.right, level+1)
	        dfsTraversal(node.left,  level+1)
        
        dfsTraversal() 
        return res